core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -4021,19 +4021,30 @@ private void validateModality(SqlNode query) {
       default:
         break;
       }
+    } else if (query.getKind() == SqlKind.WITH) {
```

Comment:
This would open the door to support `WITH` in streaming, then it would be great to have some tests for this.

Reply:
My intention is not to open support for `WITH` in streaming. I would like to fix the bug that `WITH` is not working together with `UNION`.

Reply:
I am not really familiar with Calcite core development. So it will take me quite a bit of time to add tests for streaming here.

Reply:
It's a nice-to-have, not a must-have request, I'm ok with this.